halo/evmengine/keeper/keeper_internal_test.go

```diff
@@ -0,0 +1,177 @@
+package keeper
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/omni-network/omni/halo/comet"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/k1util"
+
+	k1 "github.com/cometbft/cometbft/crypto/secp256k1"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	cmttypes "github.com/cometbft/cometbft/types"
+
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	fuzz "github.com/google/gofuzz"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeeper_isNextProposer(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		header         cmtproto.Header
+		validatorsFunc func(context.Context, int64) (*cmttypes.ValidatorSet, bool, error)
+		current        int
+		next           int
+	}
+	tests := []struct {
+		name       string
+		args       args
+		want       bool
+		wantHeight uint64
+		wantErr    bool
+	}{
+		{
+			name: "is next proposer",
+			args: args{
+				header:  cmtproto.Header{Height: 1},
```

Comment:
use a common variable to reduce duplication add a `header1 := cmtproto.Header{Height: 1}` above and use that.